paasta_tools/vitesscluster_tools.py

```diff
@@ -15,6 +17,594 @@
+    vttablet_extra_flags["queryserver-config-strict-table-acl"] = "true"
+    vttablet_extra_flags["db-credentials-server"] = "vault"
+    vttablet_extra_flags[
+        "db-credentials-vault-addr"
+    ] = f"https://vault-dre.{region}.yelpcorp.com:8200"
+    vttablet_extra_flags[
+        "db-credentials-vault-path"
+    ] = "secrets/vitess/vt-tablet/vttablet_credentials.json"
+    vttablet_extra_flags[
+        "db-credentials-vault-tls-ca"
+    ] = f"/etc/vault/all_cas/acm-privateca-{region}.crt"
+    vttablet_extra_flags["db-credentials-vault-ttl"] = "60s"
+
+    if tablet_type == "primary":
+        type = "externalmaster"
+    else:
+        type = "externalreplica"
+
+    config = {
+        "cell": cell,
+        "name": f"{db_name}_{tablet_type}",
+        "type": type,
+        "affinity": build_affinity_spec(paasta_pool),
+        "extraLabels": build_extra_labels(paasta_pool, paasta_cluster),
+        "extraEnv": build_extra_env(paasta_cluster),
+        "extraVolumeMounts": [
+            {
+                "mountPath": "/etc/vault/all_cas",
+                "name": "vault-secrets",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/vitess_keyspace_acls",
+                "name": "acls",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "etc/credentials.yaml",
+                "name": "vttablet-fake-credentials",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/init_db.sql",
+                "name": "keyspace-fake-init-script",
+                "readOnly": True,
+            },
+        ],
+        "extraVolumes": [
+            {"name": "vault-secrets", "hostPath": {"path": "/nail/etc/vault/all_cas"}},
+            {
+                "name": "acls",
+                "hostPath": {"path": "/nail/srv/configs/vitess_keyspace_acls"},
+            },
+            {"name": "vttablet-fake-credentials", "hostPath": {"path": "/dev/null"}},
+            {"name": "keyspace-fake-init-script", "hostPath": {"path": "/dev/null"}},
+        ],
+        "replicas": 1,
+        "vttablet": {
+            "extraFlags": vttablet_extra_flags,
+            "resources": {
+                "requests": {
+                    "cpu": "100m",
+                    "memory": "256Mi",
+                },
+                "limits": {"memory": "256Mi"},
+            },
+        },
+        "externalDatastore": {
+            "database": db_name,
+            "host": SOURCE_DB_HOST,
+            "port": port,
+            "user": "vt_app",
+            "credentialsSecret": {
+                "key": "/etc/credentials.yaml",
+                "volumeName": "vttablet-fake-credentials",
+            },
+        },
+        "dataVolumeClaimTemplate": {
+            "accessModes": ["ReadWriteOnce"],
+            "resources": {"requests": {"storage": "10Gi"}},
+            "storageClassName": "ebs-csi-gp3",
```

Comment:
(we'll want to make sure that this storage class exists in all our clusters)

Reply:
Is there a way we can ensure this?

Reply:
we'll probably want to add it in our flux repo 

EDIT: looks like it's already there as a common import :)